### DISCUSSION TITLE
Validation step more freqent than after 1 epoch

### DISCUSSION THREAD

**Discussion by maxiuw** [open-mmlab/mmdetection3d#3017] [General]
I want my itereation step/hook to run after 100 samples/iters. I tried different stuff 
```

# runtime settings
runner = dict(type='EpochBasedRunner', max_epochs=24)
evaluation = dict(interval=24) # change this to 1 to run validation every epoch
```
I also tried to add/modify validation interval in train_cfg to higer and lower value in combination with the evaluation value but nothing worked.
`train_cfg = dict(by_epoch=True, max_epochs=20, val_interval=250)`

I tried to modify it on BEVFusion config in `project/BEVFusion` I woul appreciate some help/directions how to do it :)

**Comment by abubake** [comment]
I believe val_interval is at a per epoch rate, so if you use 250 I think it won't validate at all. 

train_cfg = dict(by_epoch=True, max_epochs=20, val_interval=1) I believe this would validate every epoch.

**Reply by maxiuw** [reply]
So how to to validate e.g. every 10% of the epoch ? I think an option can be iterbaseloop instead of epoch based but I would like to achieve it in epoch based loop...
